src/Parsers/ExpressionElementParsers.cpp

```diff
@@ -1578,23 +1613,17 @@ bool ParserNumber::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
 
     /// try to use more exact type: UInt64
 
-    char * pos_integer = buf;
-
-    errno = 0;
-    UInt64 uint_value = std::strtoull(buf, &pos_integer, 0);
-    if (pos_integer == pos_double && errno != ERANGE && (!negative || uint_value <= (1ULL << 63)))
+    if (parseNumber(buf, pos->size(), negative, 0, res))
     {
-        if (negative)
-            res = static_cast<Int64>(-uint_value);
-        else
-            res = uint_value;
+        auto literal = std::make_shared<ASTLiteral>(res);
+        literal->begin = literal_begin;
+        literal->end = ++pos;
+        node = literal;
+
+        return true;
     }
 
-    auto literal = std::make_shared<ASTLiteral>(res);
-    literal->begin = literal_begin;
-    literal->end = ++pos;
-    node = literal;
-    return true;
+    return false;
```

Comment:
Why in changed code we always returned `true`? Was it a mistake? Maybe it's the reason of test failures.

Reply:
Yes it is issue.